pkg/injector/patch.go

```diff
@@ -138,41 +139,68 @@ func addContainer(target, add []corev1.Container, basePath string) (patch []JSON
 			path += "/-"
 		}
 		patch = append(patch, JSONPatchOperation{
-			Op:    "add",
+			Op:    addOperation,
 			Path:  path,
 			Value: value,
 		})
 	}
 	return patch
 }
 
-func updateAnnotation(target, add map[string]string, basePath string) (patch []JSONPatchOperation) {
-	for key, value := range add {
-		if target == nil {
-			// First one will be a Create
-			target = map[string]string{}
-			patch = append(patch, JSONPatchOperation{
-				Op:   "add",
-				Path: basePath,
-				Value: map[string]string{
-					key: value,
-				},
-			})
-		} else {
-			// Update
-			op := "add"
-			if target[key] != "" {
-				op = "replace"
-			}
-			patch = append(patch, JSONPatchOperation{
-				Op:    op,
-				Path:  path.Join(basePath, escapeJSONPointerValue(key)),
-				Value: value,
-			})
+func createTarget(key, path, value string) JSONPatchOperation {
+	return JSONPatchOperation{
+		Op:   addOperation,
+		Path: path,
+		Value: map[string]string{
+			key: value,
```

Comment:
Function is very specific to patching map types in API objects, so I would want to include the type (map) in the function name.